tests/pubsub/check_pubsub_get_state.c

```diff
@@ -0,0 +1,437 @@
+
+    UA_NodeId DsWId_Conn1_WG1_DS1;
+    UA_NodeId_init(&DsWId_Conn1_WG1_DS1);
+    UA_NodeId VarId_Conn1_WG1_DS1;
+    UA_NodeId_init(&VarId_Conn1_WG1_DS1);
+    UA_NodeId PDSId_Conn1_WG1_PDS1;
+    UA_NodeId_init(&PDSId_Conn1_WG1_PDS1);
+    UA_UInt32 DSWNo_Conn1_WG1_DS1 = 1;
+    AddPublishedDataSet(&WGId_Conn1_WG1, "Conn1_WG1_PDS1", "Conn1_WG1_DS1", DSWNo_Conn1_WG1_DS1, &PDSId_Conn1_WG1_PDS1, 
+        &VarId_Conn1_WG1_DS1, &DsWId_Conn1_WG1_DS1);
+
+    /* setup Connection 1: reader */
+    UA_NodeId RGId_Conn1_RG1;
+    UA_NodeId_init(&RGId_Conn1_RG1);
+    AddReaderGroup(&ConnId_1, "Conn1_RG1", &RGId_Conn1_RG1);
+
+    UA_NodeId DSRId_Conn1_RG1_DSR1;
+    UA_NodeId_init(&DSRId_Conn1_RG1_DSR1);
+    UA_NodeId VarId_Conn1_RG1_DSR1;
+    UA_NodeId_init(&VarId_Conn1_RG1_DSR1);
+    UA_Duration MessageReceiveTimeout_Conn1_RG1_DSR1 = 350.0;
+    AddDataSetReader(&RGId_Conn1_RG1, "Conn1_RG1_DSR1", PublisherNo_Conn1, WGNo_Conn1_WG1, DSWNo_Conn1_WG1_DS1, 
+        MessageReceiveTimeout_Conn1_RG1_DSR1, &VarId_Conn1_RG1_DSR1, &DSRId_Conn1_RG1_DSR1);
+
+
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "check state");
+    UA_PubSubState state = UA_PUBSUBSTATE_ERROR;
+
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_WriterGroup_getState(server, WGId_Conn1_WG1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_DISABLED, state);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_DataSetWriter_getState(server, DsWId_Conn1_WG1_DS1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_DISABLED, state);
+
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_ReaderGroup_getState(server, RGId_Conn1_RG1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_DISABLED, state);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_DataSetReader_getState(server, DSRId_Conn1_RG1_DSR1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_DISABLED, state);
+
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "set groups operational");
+
+    ck_assert(UA_Server_setWriterGroupOperational(server, WGId_Conn1_WG1) == UA_STATUSCODE_GOOD);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_WriterGroup_getState(server, WGId_Conn1_WG1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_OPERATIONAL, state);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_DataSetWriter_getState(server, DsWId_Conn1_WG1_DS1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_OPERATIONAL, state);
+
+    ck_assert(UA_Server_setReaderGroupOperational(server, RGId_Conn1_RG1) == UA_STATUSCODE_GOOD);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_ReaderGroup_getState(server, RGId_Conn1_RG1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_OPERATIONAL, state);
+    ck_assert_int_eq(UA_STATUSCODE_GOOD, UA_Server_DataSetReader_getState(server, DSRId_Conn1_RG1_DSR1, &state));
+    ck_assert_int_eq(UA_PUBSUBSTATE_OPERATIONAL, state);
+
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "set groups disabled");
+    ck_assert(UA_Server_setWriterGroupDisabled(server, WGId_Conn1_WG1) == UA_STATUSCODE_GOOD);
```

Comment:
In the long term we should think about changing our API. Now we have a getState, maybe it would be better if we change this function also to setState with the target state as argument (not for this PR).

Reply:
ok
may I resolve the conversation?